pkg/internal/launcher/launcher.go

```diff
@@ -74,8 +76,10 @@ func (l Launcher) Run(ctx context.Context) (runErr error) {
 	}
 
 	defer func() {
-		if err := l.checkup.Teardown(context.Background()); err != nil {
-			runStatus.FailureReason = append(runStatus.FailureReason, err.Error())
+		if !l.shouldSkipTeardown(runStatus.FailureReason) {
+			if err := l.checkup.Teardown(context.Background()); err != nil {
```

Comment:
It's just cosmetics, but I think less tabs + less nots + early return is better:
```go
if l.shouldSkipTeardown(runStatus.FailureReason) {
  return
}
```

Reply:
We could argue the current implementation is more self-contained (you can't skip teardow, do teardown) but I'm ok with the change. ACK.